### PR TITLE
Tooltip: make keyboard dismissable

### LINKED ISSUE #9435: WCAG 2.1: 1.4.13: Tooltip component is not keyboard-dismissable
Migrated from https://github.com/microsoft/accessibility-insights-web/issues/777

## Codepen demonstrating the issue

https://codepen.io/dbjorge/pen/QXbvzN (this is just an export of the default Tooltip example from https://developer.microsoft.com/en-us/fabric#/controls/web/tooltip as of writing)

## Actual behavior:

* Go to the codepen
* Click the "Hover Over Me" element
* Press "Tab" then "Shift+Tab" to simulate keyboard navigating to element
* Wait for tooltip to appear
* Press "Esc"
* Observe that tooltip does not disappear

## Expected behavior:

tooltip should be dismissable using esc key

## Documentation describing expected behavior

From https://www.w3.org/TR/WCAG21/#content-on-hover-or-focus

> Success Criterion 1.4.13 Content on Hover or Focus
> Where receiving and then removing pointer hover or keyboard focus triggers additional content to become visible and then hidden, the following are true:
> ...
> Dismissable
> A mechanism is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an input error or does not obscure or replace other content;

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #9435 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

These changes allow Tooltip to be keyboard dismissable. This aligns with the W3C spec: https://www.w3.org/TR/WCAG21/#content-on-hover-or-focus

Note: most of the file changes are snapshot updates that include the new `onKeyDown`.

#### Focus areas to test

Ensure that when tabbing to a tooltiphost, the tooltip can be dismissed by pressing the escape key. Pressing the escape key on a tooltiphost in a panel should dismiss the tooltip and the panel.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9470)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.base.tsx [hunk 1]
```diff
@@ -79,6 +80,7 @@ export class TooltipHostBase extends BaseComponent<ITooltipHostProps, ITooltipHo
         {...{ onBlurCapture: this._hideTooltip }}
         onMouseEnter={this._onTooltipMouseEnter}
         onMouseLeave={this._onTooltipMouseLeave}
+        onKeyDown={this._onTooltipKeyDown}
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.base.tsx]: does this onKeyDown need to be on both the div and the Tooltip and the callout?

Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.base.tsx]: Actually it looks like we might just need it on the outermost div. I'll push the changes

**Comment by KevinTCoughlin** [comment]
@natalieethell, when I hover over the example at: http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/9470/merge/fabric-website-resources/dist/index.html#/examples/Tooltip

A run-time error is thrown. Is it related to your change-set?

**Comment by natalieethell** [comment]
@KevinTCoughlin oh, good find. Looks like it's happening on the master branch, too, so it must be unrelated to this change: http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/heads/master/fabric-website-resources/dist/index.html#/examples/Tooltip.

**Comment by natalieethell** [comment]
Found the root of the cause - https://github.com/OfficeDev/office-ui-fabric-react/pull/9341. The tooltip seems to be working as expected in 7.2.0, but I'm seeing this run-time error in 7.3.0, which has those tooltip perf changes.

**Comment by KevinTCoughlin** [comment]
> Found the root of the cause - #9341. The tooltip seems to be working as expected in 7.2.0, but I'm seeing this run-time error in 7.3.0, which has those tooltip perf changes.

@aneeshack4 do we have to prioritize a fix for the above?

**Comment by aneeshack4** [comment]
@KevinTCoughlin @natalieethell weird I see the repro on that link to the master branch but I don't see it reproing on the website https://developer.microsoft.com/en-us/fabric#/controls/web/tooltip The website's not on 7.3.0 yet? As for prioritization, it was a perf improvement addressing a issue that came in while I was on shield so I don't think it's the highest priority but important. Looping in @micahgodbolt, component owner of tooltip - Micah, what do you think?

**Comment by natalieethell** [comment]
Reverting here: https://github.com/OfficeDev/office-ui-fabric-react/pull/9495

**Comment by natalieethell** [comment]
Unrelated a11y-tests are failing, which I think may be from the changes in this PR: https://github.com/OfficeDev/office-ui-fabric-react/pull/9479

**Comment by micahgodbolt** [comment]
oh goodie....time to see how these tests are working @statm

**Comment by statm** [comment]
Looks like some random puppeteer glitch. Passed now.

**Comment by natalieethell** [comment]
@micahgodbolt should be working again if you could take a look. Thanks!

**Comment by micahgodbolt** [comment]
we've got enough signoffs here i'm going to merge this in.

**Pull request merged**